graphql-dgs-spring-graphql/src/main/kotlin/com/netflix/graphql/dgs/springgraphql/DgsGraphQLSourceBuilder.kt

```diff
@@ -0,0 +1,164 @@
+import graphql.GraphQL
+import graphql.execution.instrumentation.ChainedInstrumentation
+import graphql.execution.instrumentation.Instrumentation
+import graphql.schema.*
+import graphql.schema.GraphQLSchema.BuilderWithoutTypes
+import graphql.schema.idl.RuntimeWiring
+import graphql.schema.idl.TypeDefinitionRegistry
+import org.springframework.core.io.Resource
+import org.springframework.graphql.execution.*
+import org.springframework.graphql.execution.GraphQlSource.SchemaResourceBuilder
+import org.springframework.lang.Nullable
+import java.util.*
+import java.util.function.BiFunction
+import java.util.function.Consumer
+
+class DgsGraphQLSourceBuilder(private val dgsSchemaProvider: DgsSchemaProvider, private val reloadSchemaIndicator: DefaultDgsQueryExecutor.ReloadSchemaIndicator) : SchemaResourceBuilder {
+    private val typeDefinitionConfigurers = mutableListOf<TypeDefinitionConfigurer>()
+    private val runtimeWiringConfigurers = mutableListOf<RuntimeWiringConfigurer>()
+
+    private val schemaResources: Set<Resource> = LinkedHashSet()
+    private val exceptionResolvers = mutableListOf<DataFetcherExceptionResolver>()
+    private val subscriptionExceptionResolvers = mutableListOf<SubscriptionExceptionResolver>()
+    private val typeVisitors = mutableListOf<GraphQLTypeVisitor>()
+    private val typeVisitorsToTransformSchema = mutableListOf<GraphQLTypeVisitor>()
+    private val instrumentations = mutableListOf<Instrumentation>()
+    private var graphQlConfigurers = Consumer { builder: GraphQL.Builder -> }
+
+    @Nullable
+    private var typeResolver: TypeResolver? = null
+
+    @Nullable
+    private var schemaReportConsumer: Consumer<SchemaReport>? = null
+
+    override fun exceptionResolvers(resolvers: List<DataFetcherExceptionResolver>): SchemaResourceBuilder {
+        exceptionResolvers.addAll(resolvers)
+        return this
+    }
+
+    override fun subscriptionExceptionResolvers(resolvers: List<SubscriptionExceptionResolver>): SchemaResourceBuilder {
+        subscriptionExceptionResolvers.addAll(resolvers)
+        return this
+    }
+
+    override fun typeVisitors(typeVisitors: List<GraphQLTypeVisitor>): SchemaResourceBuilder {
+        this.typeVisitors.addAll(typeVisitors)
+        return this
+    }
+
+    override fun typeVisitorsToTransformSchema(typeVisitors: List<GraphQLTypeVisitor>): SchemaResourceBuilder {
+        typeVisitorsToTransformSchema.addAll(typeVisitors)
+        return this
+    }
+
+    override fun instrumentation(instrumentations: List<Instrumentation>): SchemaResourceBuilder {
+        this.instrumentations.addAll(instrumentations)
+        return this
+    }
+
+    override fun configureGraphQl(configurer: Consumer<GraphQL.Builder>): SchemaResourceBuilder {
+        graphQlConfigurers = this.graphQlConfigurers.andThen(configurer)
+        return this
+    }
+
+    override fun build(): GraphQlSource {
+        return ReloadableGraphQLSource(reload(), this, reloadSchemaIndicator)
```

Comment:
One specific thing, if you are to extend `AbstractGraphQlSourceBuilder` instead, obviously that creates the very simple `FixedGraphQlSource` while what you want is one that resolves at runtime, possibly by reloading. This can be achieved by changing `ReloadableGraphQLSource` a little to be created with a `GraphQlSource.Builder` rather than with a `GraphQlSource`. That way it can call build again whenever it wants. 

This should be a fairly small refactoring that will keep the builder focused on building a single instance, and `ReloadableGraphQlSource` responsible for all that has to do with reloading. We could also consider adding a protected method to create the `GraphQlSource` instance if needed.